src/plugins/intel_cpu/src/nodes/gather_compression.cpp

```diff
@@ -0,0 +1,210 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gather_compression.h"
+
+#include "common/cpu_memcpy.h"
+#include "transformations/cpu_opset/common/op/gather_compression.hpp"
+#include "utils/ngraph_utils.hpp"
+
+using namespace dnnl::impl::cpu;
+
+#define THROW_ERROR(...) OPENVINO_THROW(getTypeStr(), " node with name '", getName(), "' ", __VA_ARGS__)
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+bool GatherCompression::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto gather_compression = std::dynamic_pointer_cast<const GatherCompressionNode>(op);
+        if (!gather_compression) {
+            errorMessage = "Only GatherCompression operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+GatherCompression::GatherCompression(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr context)
+    : Node(op, context, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)) {
```

Comment:
@usstq , about "NgraphShapeInferFactory", I don't implement my own shape infer, it will fallback to "GatherCompressionNode::validate_and_infer_types", is it OK?